tests/data/test_model.py

```diff
@@ -0,0 +1,83 @@
+"""Test for ModelData class"""
+
+from pathlib import Path
+import shutil
+
+from aiida_mlip.data.model import ModelData
+
+
+def test_local_file():
+    """Testing that the local file function works"""
+    # Construct a ModelData instance with the local file
+    model_path = Path("./tests/input_files/model_local_file.txt")
+    model = ModelData.local_file(model_path)
+    # Assert the ModelData contains the content we expect
+    content = model.get_content()
+    assert content == model_path.read_text(encoding="utf-8")
+
+
+def test_architecture():
+    """Testing that the architecture is read and added to attributes"""
+    model = ModelData.local_file(
+        "./tests/input_files/model_local_file.txt",
+        filename="model",
+        architecture="mace",
+    )
+    assert model.architecture == "mace"
+
+
+def test_download_fresh_file():
+    """Test if download works"""
+    # Ensure we do not have the file cached already
+    path_test = Path("./tests/data/tmp/mace/test_download.txt")
+    path_test.unlink(missing_ok=True)
+
+    # Construct a ModelData instance downloading a non-cached file
+    model = ModelData.download(
+        url="https://raw.githubusercontent.com/stfc/aiida-mlip/main/tests/input_files/file2.txt",
+        filename="test_download.txt",
+        cache_dir="./tests/data/tmp",
+        architecture="mace",
+    )
+
+    # Assert the ModelData contains the content we expect
+    content = model.get_content()
+    assert content == "file with content\ncontent2\n"
+    file_path = Path("./tests/data/tmp/mace/test_download.txt")
+    assert file_path.exists(), f"File {file_path} does not exist."
+    # Clear test cache
+    shutil.rmtree(Path("./tests/data/tmp/mace/"))
```

Comment:
Would it work to use pytest's [tmpdir](https://docs.pytest.org/en/6.2.x/tmpdir.html) rather than making/deleting our own?

Reply:
Or Python stdlib's [tempfile](https://docs.python.org/3/library/tempfile.html)